### PR TITLE
chore: make libxml optional

### PR DESCRIPTION
**Pull request by Antonov548**
Closes https://github.com/igraph/rigraph/issues/1217

### CODE REVIEW

**Review by krlmlr — APPROVED** [review]
Thanks! Can you please update the `SystemRequirements` section in `DESCRIPTION` too, also for GLPK?

File: configure [hunk 1]
```diff
@@ -8,24 +8,21 @@ elif pkg-config --version >/dev/null 2>&1; then
     libxml2_cflags=$(pkg-config --cflags libxml-2.0)
     libxml2_libs=$(pkg-config --libs --static libxml-2.0)
 else
-    echo "Error: Neither xml2-config nor pkg-config is available"
-    exit 1
+    echo "Warning: libxml not found. Neither xml2-config nor pkg-config is available"
```
Comment (szhorvat) [inline configure:11]:
```suggestion
    echo "Warning: libxml2 not found. Neither xml2-config nor pkg-config is available."
```

File: configure [hunk 2]
```diff
@@ -8,24 +8,21 @@ elif pkg-config --version >/dev/null 2>&1; then
...
 fi
 
 if [ -n "$libxml2_cflags" ]; then
-    PKG_CFLAGS="$PKG_CFLAGS $libxml2_cflags"
+    PKG_CFLAGS="$PKG_CFLAGS $libxml2_cflags -DHAVE_LIBXML"
     echo "libxml2 include directories: $libxml2_cflags"
 else
-    echo "Error: libxml2 include directory not found"
-    exit 1
+    echo "Warning: libxml2 include directory not found"
```
Comment (szhorvat) [inline configure:18]:
```suggestion
    echo "Warning: libxml2 include directory not found."
```

File: configure [hunk 3]
```diff
@@ -8,24 +8,21 @@ elif pkg-config --version >/dev/null 2>&1; then
...
 fi
 
 if [ -n "$libxml2_libs" ]; then
     PKG_LIBS="$PKG_LIBS $libxml2_libs"
     echo "libxml2 library link flags: $libxml2_libs"
 else
-    echo "Error: libxml2 library link flags not found"
-    exit 1
+    echo "Warning: libxml2 library link flags not found"
```
Comment (szhorvat) [inline configure:25]:
```suggestion
    echo "Warning: libxml2 library link flags not found."
```

**Review by szhorvat — APPROVED** [review]
If libxml2 detection fails at either of the three steps, could we print a message saying something like "GraphML import will not be available." ?

**Comment by szhorvat** [comment]
Also, it would be good if someone could test this on a system where libxml2 is not available: make sure it compiles, links, and trying to import a GraphML file gives a nice error. I don't see any reason why it wouldn't work, but just in case ... it's hard for me to hide libxml2 since it comes as standard on macOS.

**Review by szhorvat — APPROVED** [review]

**Pull request merged**